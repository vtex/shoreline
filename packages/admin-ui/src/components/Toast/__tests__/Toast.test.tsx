import React from 'react'
import { render, axe } from '../../../test-utils'

import { ToastProvider } from '../index'
import { Toast } from '../components/Toast'

const toastProps = {
  onClear: () => {},
  id: 'toast-id',
  dedupeKey: 'toast-id',
  message: 'This is a toast!',
  action: {
    label: 'Action',
    onClick: () => {},
  },
  shouldRemove: true,
  dismissible: true,
}

describe('Toast', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(
      <ToastProvider>
        <Toast {...toastProps} />
      </ToastProvider>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
