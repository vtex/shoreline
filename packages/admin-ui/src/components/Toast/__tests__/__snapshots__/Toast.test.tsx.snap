// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should match snapshot 1`] = `
<DocumentFragment>
  .emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: subtle;
  border: 1px solid hsla(206, 72%, 70%, 1);
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #000000;
  border-radius: 4px;
  background: hsla(206, 61%, 92%, 1);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
  color: hsla(206, 63%, 44%, 1);
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
}

.emotion-5:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-5:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-5:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-5 svg {
  margin: 0.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: #000000;
  background: transparent;
}

.emotion-9:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-9:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-9:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-9 svg {
  margin: 0.25rem;
}

.emotion-6 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="currentColor"
        height="1.25rem"
        viewBox="0 0 256 256"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <path
          d="M56.20305,104A71.899,71.899,0,0,1,128.5484,32.002c39.58967.29432,71.25651,33.20133,71.25651,72.90185V112c0,35.81563,7.49325,56.59893,14.093,67.95814A7.999,7.999,0,0,1,207.01628,192H48.98365A7.99908,7.99908,0,0,1,42.103,179.95641c6.60328-11.35959,14.1-32.1426,14.1-67.95641Z"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
        />
        <path
          d="M96,192v8a32,32,0,0,0,64,0v-8"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="18"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-6"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="56"
                y2="200"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="200"
                y2="56"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
}

.emotion-5:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-5:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-5:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-5 svg {
  margin: 0.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: #000000;
  background: transparent;
}

.emotion-9:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-9:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-9:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-9 svg {
  margin: 0.25rem;
}

.emotion-6 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: subtle;
  border: 1px solid hsla(123, 46%, 67%, 1);
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #000000;
  border-radius: 4px;
  background: hsla(123, 39%, 91%, 1);
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
  color: hsla(123, 41%, 37%, 1);
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="currentColor"
        height="1.25rem"
        viewBox="0 0 256 256"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <path
          d="M128,24A104,104,0,1,0,232,128,104.12041,104.12041,0,0,0,128,24Zm49.53125,85.78906-58.67187,56a8.02441,8.02441,0,0,1-11.0625,0l-29.32813-28a8.00675,8.00675,0,0,1,11.0625-11.57812l23.79687,22.72656,53.14063-50.72656a8.00675,8.00675,0,0,1,11.0625,11.57812Z"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-6"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="56"
                y2="200"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="200"
                y2="56"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
}

.emotion-5:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-5:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-5:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-5 svg {
  margin: 0.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: #000000;
  background: transparent;
}

.emotion-9:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-9:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-9:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-9 svg {
  margin: 0.25rem;
}

.emotion-6 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: subtle;
  border: 1px solid hsla(28, 87%, 70%, 1);
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #000000;
  border-radius: 4px;
  background: hsla(30, 100%, 92%, 1);
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
  color: hsla(30, 100%, 42%, 1);
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="currentColor"
        height="1.25rem"
        viewBox="0 0 256 256"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <path
          d="M236.74414,187.96585l-87.96875-152a24.012,24.012,0,0,0-41.54687,0v.00782L19.25977,187.96585a23.9982,23.9982,0,0,0,20.76562,36.02344H215.97852a23.9982,23.9982,0,0,0,20.76562-36.02344Zm-116.75-83.96875a8,8,0,1,1,16,0v40a8,8,0,1,1-16,0ZM128,192a12,12,0,1,1,12-12A12.00059,12.00059,0,0,1,128,192Z"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-6"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="56"
                y2="200"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="200"
                y2="56"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 >*:first-child {
  margin-right: 0.75rem;
}

.emotion-1 {
  text-align: start;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
}

.emotion-5:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-5:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-5:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-5 svg {
  margin: 0.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: 32px;
  padding-left: 0.125rem;
  padding-right: 0.125rem;
  color: #000000;
  background: transparent;
}

.emotion-9:disabled {
  background: hsla(0, 0%, 90%, 1);
  color: hsla(0, 0%, 45%, 1);
}

.emotion-9:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-9:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-9:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-9:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-9 svg {
  margin: 0.25rem;
}

.emotion-6 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 16.125rem;
  width: auto;
  min-height: 4.5rem;
  height: auto;
  max-height: 4.5rem;
  padding: 1rem;
  box-shadow: subtle;
  border: 1px solid hsla(0, 81%, 77%, 1);
  pointer-events: all;
  overflow: hidden;
  padding-top: 1rem;
  padding-bottom: 1rem;
  color: #000000;
  border-radius: 4px;
  background: hsla(0, 60%, 93%, 1);
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
  color: hsla(0, 58%, 52%, 1);
}

<div
    class="emotion-11"
    role="alert"
  >
    <div
      class="emotion-2"
    >
      <svg
        class="emotion-0"
        fill="currentColor"
        height="1.25rem"
        viewBox="0 0 256 256"
        width="1.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="16rem"
          width="16rem"
        />
        <path
          clip-rule="evenodd"
          d="M175.766 28.6875L227.312 80.2344C230.303 83.2416 231.987 87.3064 232 91.5472V164.453C231.987 168.694 230.303 172.758 227.312 175.766L175.766 227.313C172.759 230.302 168.694 231.987 164.453 232H91.5466C87.3058 231.987 83.2418 230.302 80.2346 227.313L28.6874 175.766C25.6974 172.758 24.0131 168.694 24 164.453V91.5472C24.0131 87.3064 25.6974 83.2416 28.6874 80.2344L80.2346 28.6875C83.2418 25.6974 87.3058 24.0132 91.5466 24H164.453C168.694 24.0132 172.759 25.6974 175.766 28.6875ZM167.387 156.932C166.984 155.962 166.393 155.079 165.649 154.337L139.306 127.993L165.649 101.65C167.147 100.149 167.987 98.1152 167.986 95.9952C167.985 93.8752 167.143 91.8416 165.643 90.3432C164.144 88.844 162.111 88.0008 159.991 88C157.871 87.9992 155.838 88.8392 154.337 90.3368L127.993 116.681L101.649 90.3368C100.149 88.8392 98.1146 87.9992 95.9946 88C93.8746 88.0008 91.8418 88.844 90.3426 90.3432C88.8434 91.8416 88.001 93.8752 88.0002 95.9952C87.9986 98.1152 88.8394 100.149 90.337 101.65L116.68 127.993L90.337 154.337C88.8394 155.838 87.9986 157.871 88.0002 159.991C88.001 162.111 88.8434 164.144 90.3426 165.643C91.8418 167.142 93.8746 167.985 95.9946 167.986C98.1146 167.987 100.149 167.147 101.649 165.65L127.993 139.306L154.337 165.65C155.079 166.394 155.961 166.984 156.932 167.387C157.903 167.79 158.943 167.998 159.995 167.998C161.046 167.999 162.087 167.792 163.059 167.39C164.03 166.988 164.912 166.398 165.655 165.655C166.399 164.912 166.988 164.03 167.39 163.058C167.792 162.087 167.999 161.046 167.999 159.995C167.998 158.944 167.79 157.903 167.387 156.932Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <p
        class="emotion-1"
      >
        This is a toast!
      </p>
    </div>
    <div
      class="emotion-10"
    >
      <button
        class="emotion-5"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
             Action
          </div>
        </div>
      </button>
      <button
        aria-label="Close toast"
        class="emotion-9"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-3"
          >
            <svg
              class="emotion-6"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="56"
                y2="200"
              />
              <line
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="200"
                x2="56"
                y1="200"
                y2="56"
              />
            </svg>
             
          </div>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
